### ISSUE TITLE
'IPsec Settings...' button greyed out in Plasma L2TP networkmanager GUI

### ISSUE THREAD

**Issue by AchmadFathoni** [NixOS/nixpkgs#336660]
### Describe the bug
When adding L2TP VPN connection in Plasma networkmanager, I can't set IPsec password since the relevant button is greyed out. This bug affect other user as in [here](https://discourse.nixos.org/t/ipsec-setting-on-kde-plasma/17342).

I think I found the trouble. The current nixpkg's path [patch](https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/tools/networking/networkmanager/l2tp/fix-paths.patch) is no longer applicable for networkmanager L2TP 1.20.16 since [utils.c](https://github.com/nm-l2tp/NetworkManager-l2tp/blob/main/shared/utils.c) is modified so `nm_find_ipsec` & `nm_find_l2tpd` is now in another line number.

### Steps To Reproduce
1. Put `networking.networkmanager.enableStrongSwan = true;` in configuration.nix
2. Open **Wi-Fi & Networking** menu in **System Settings** window
3. Open create **Layer 2 Tunneling Protocol (L2TP)** window

### Expected behavior
After enabled the network manager strongswan, this button should be available.

### Screenshots
![Example](https://discourse.nixos.org/uploads/default/original/2X/5/5ace824da28297b547f0e6c5e5ac3a461af8242b.png)

### Notify maintainers
@abbradar @obadz

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.40, NixOS, 24.05 (Uakari), 24.05.20240717.c716603`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"home-manager-24.05.tar.gz, nixos-24.05"`
 - nixpkgs: `/nix/store/sj9yrq21wbbfr5715hys3laa2qd6x471-source
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by dkosovic** [comment]
The issue isn't with NetworkManager-l2tp's `utils.c` patch as that is only for the GTK3 and GTK4 based L2TP editor plugins.

Extract from plasma-nm's `l2tpipsecwidget.cpp` which has a `L2tpIpsecWidget::hasIpsecDaemon()` function:
- https://invent.kde.org/plasma/plasma-nm/-/blame/v6.1.4/vpn/l2tp/l2tpipsecwidget.cpp?ref_type=tags#L334

```C++
QString ipsecBinary = QStandardPaths::findExecutable(QStringLiteral("ipsec"), QStringList()
     << QStringLiteral("/sbin") << QStringLiteral("/usr/sbin"));

// On some Linux distributions, ipsec executable has been renamed strongswan
if (ipsecBinary.isEmpty()) {
    ipsecBinary = QStandardPaths::findExecutable(QStringLiteral("strongswan"), QStringList() <<
         << QStringLiteral("/sbin") << QStringLiteral("/usr/sbin"));
}
```

Without a patch, plasma-nm only looks for `/sbin/ipsec`, `/usr/sbin/ipsec`, `/sbin/strongswan` or `/usr/sbin/strongswan` to determine if there is an IPsec daemon.

**Comment by AchmadFathoni** [comment]
Yes it's actually works, thank you @dkosovic. I made  https://github.com/NixOS/nixpkgs/commit/9e4846033d127b5c6df66d8488db3fb618710b7b and will open a PR here when I got time.

**Comment by benibaumi** [comment]
I ran into the same issue on NixOS, where Plasma NM can’t find the ipsec binary because it only searches /sbin and /usr/sbin, as already mentioned above.

My tmp workaround: I fixed it by adding a tmpfiles rule that creates a symlink to the StrongSwan binary from the Nix store in my configuration.nix:

`systemd.tmpfiles.rules = [
    "L+ /usr/sbin/ipsec - - - - ${pkgs.strongswan}/sbin/ipsec"
  ];`

After rebuilding, the IPsec Settings… button in the Plasma L2TP dialog became active and configuration worked correctly.

System info:

` - system: `"x86_64-linux"`
 - host os: `Linux 6.12.56, NixOS, 25.05 (Warbler), 25.05.812242.3de8f8d73e35`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.5`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/store/d6891wvkwmzjg2paph543qnz7ly6bi98-nixos-25.05/nixos`
`
